File: backend/app/schemas/pipeline.py
"""Pipeline-related Pydantic models."""

from __future__ import annotations

from pydantic import BaseModel


class StrategyInfoResponse(BaseModel):
    slot_name: str
    current_impl: str
    available_impls: list[str] = []
    config: dict = {}


class StageDescriptionResponse(BaseModel):
    name: str
    order: int
    category: str
    is_active: bool = True
    strategies: list[StrategyInfoResponse] = []


class PipelineDescribeResponse(BaseModel):
    name: str
    stages: list[StageDescriptionResponse]


class PresetInfoResponse(BaseModel):
    name: str
    description: str
    active_stages: list[int]


class PresetsListResponse(BaseModel):
    presets: list[PresetInfoResponse]

File: frontend/src/types/pipeline.ts
export interface StrategyInfo {
  slot_name: string;
  current_impl: string;
  available_impls: string[];
  config: Record<string, unknown>;
}

export interface StageDescription {
  name: string;
  order: number;
  category: string;
  is_active: boolean;
  strategies: StrategyInfo[];
}

export interface PipelineDescribeResponse {
  name: string;
  stages: StageDescription[];
}

export interface PresetInfo {
  name: string;
  description: string;
  active_stages: number[];
}
